src/domains/project/service/project.service.impl.ts

```diff
@@ -14,22 +14,45 @@ export class ProjectServiceImpl implements ProjectService {
     private readonly userRepository: UserRepository
   ) {}
 
+  /**
+   * Deletes a project by its ID. Checks for user's permissions for deleting projects.
+   *
+   * @param input - An object containing user ID and project ID for project deletion.
+   * @param {string} input.cognitoId - AWS Cognito ID of the user trying to delete the project.
+   * @param {string} input.projectId - ID of the project to delete.
+   * @throws {NotFoundException} - If the project with the specified ID is not found.
+   * @returns {Promise<string>} - A promise that resolves to the deleted ID upon successful deletion.
+   */
+  async deleteProject({ cognitoId, projectId }: { cognitoId: string; projectId: string }): Promise<string> {
+    const user = await this.userRepository.getByCognitoId(cognitoId);
+    if (user == null) throw new NotFoundException('User');
+
+    const project = await this.projectRepository.getProjectDetails({ projectId, userId: user.id });
+    if (project == null) throw new NotFoundException('Project');
```

Comment:
I would add a validation to check if the project has been logically deleted it should throw an Exception too.

Reply:
I don't see it necessary to perform double validation.